File: 0240-search-a-2d-matrix-ii/0240-search-a-2d-matrix-ii.js
/**
 * @param {number[][]} matrix
 * @param {number} target
 * @return {boolean}
 */
var searchMatrix = function(arr, target) {
    let n = arr.length;
    let m = arr[0].length;
    let r = n-1;
    let c = 0;
    while(r>=0 && c<m){
        let curr = arr[r][c];
        if(curr == target){
            return true;
        }
        else if(curr < target){
            c++;
        }
        else{
            r--;
        }
    }
    return false;
};

File: 0240-search-a-2d-matrix-ii/0240-search-a-2d-matrix-ii.java
class Solution {
    public boolean searchMatrix(int[][] matrix, int target) {
        int n = matrix.length;
        int row = matrix.length - 1, col = matrix[0].length - 1;
        int low = 0, high = col;
        while (low <= row && high >= 0) {
            if (matrix[low][high] == target) {
                return true;
            } else if (matrix[low][high] < target) {
                low++;
            } else {
                high--;
            }
        }
        return false;
    }
}